### PR TITLE
Add gometalinter usage notes

### PR DESCRIPTION
**Pull request by prateek**
/cc @robskillington

### CODE REVIEW

**Comment by prateek** [comment]
@jeromefroe any points to add/amend based on your experience adding the metalinter to m3collector?

File: GUIDELINE.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    @($(metalint_check) $(metalint_config) $(metalint_exclude) \
+      && echo "metalinted successfully!") || (echo "metalinter failed" && exit 1)
+  ```
+
+  - Configuration for the underlying linter, `.metalinter.json`
+
+  ```
+  {
+    "Linters": {
+      "unused":   "unused -tags integration:PATH:LINE:COL:MESSAGE",
```
Comment (jeromefroe) [inline GUIDELINE.md]: Might add a note here about the build tags.

File: GUIDELINE.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+### Q&A
+- Which linters to enable?
+`gometalinter` integrates with 10-15 linters, some which share concerns. I suggest enabling them one at a time, and work through the issues raised by
+each.
+
+In some cases, you'll find there's too much noise (especially with `dupl`, and `gocyclo`). Skim the issues raised and if you don't find anything useful,
+skip enabling that particular linter all together.
+
+In other cases, you may find there are certain errors (especially default value for enums) which are raise as false positives. For those cases, you'll have
```
Comment (jeromefroe) [inline GUIDELINE.md]: s/raise/raised

File: GUIDELINE.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+to supress the warnings using comments. More on this in a section below.
```
Comment (jeromefroe) [inline GUIDELINE.md]: s/supress/suppress

**Comment by prateek** [comment]
@jeromefroe thanks for the feedback. switched to using m3collector as the canonical example, and put a `<details>` snippet explaining the m3db usage.

**Pull request merged**